pkg/ipmi/ipmi.go

```diff
@@ -191,6 +225,57 @@ func (i *IPMI) ShutoffWatchdog() error {
 	return nil
 }
 
+func (e *Event) marshall() ([]byte, error) {
+	buf := &bytes.Buffer{}
+
+	if err := binary.Write(buf, binary.LittleEndian, *e); err != nil {
+		log.Fatalf("Can't convert struct %v to bytes", *e)
+		return nil, err
+	}
+
+	data := make([]byte, 16)
+
+	//system event record
+	if buf.Bytes()[2] == 0x2 {
+		copy(data[:], buf.Bytes()[:16])
+	}
+
+	//OEM timestamped
+	if buf.Bytes()[2] >= 0xC0 && buf.Bytes()[2] <= 0xDF {
+		copy(data[0:3], buf.Bytes()[0:3])
+		copy(data[3:16], buf.Bytes()[16:29])
+	}
+
+	//OEM non-timestamped
+	if buf.Bytes()[2] >= 0xE0 && buf.Bytes()[2] <= 0xFF {
+		copy(data[0:3], buf.Bytes()[0:3])
+		copy(data[3:16], buf.Bytes()[29:42])
+	}
+
+	return data, nil
+}
+
+func (i *IPMI) LogSystemEvent(e *Event) error {
+	req := &req{}
+	req.msg.cmd = _BMC_ADD_SEL
+	req.msg.netfn = _IPMI_NETFN_STORAGE
+
+	data, err := e.marshall()
+
+	if err != nil {
+		return err
+	}
+
+	req.msg.data = unsafe.Pointer(&data[0])
+	req.msg.dataLen = 16
+
+	buf, err := i.sendrecv(req)
+
+	log.Println("The receive data for Log System Event is", buf)
```

Comment:
should something be interpreting the buffer and turn back into a structure that can be returned from LogSystemEvent?

Reply:
May I know the purpose for returning the buffer from LogSystemEvent?

Reply:
Oh, I was kind of asking you! You're printing it here, so it seems like it contains some useful information?

Reply:
I am sorry for confusing you, the `buf` includes `Completion Code` and `Record ID`, I think they just need to be printed out, I can print detail information for them.

Reply:
Oh I see. I'd prefer that this ipmi library not log anything or that it be optional. Would it be useful to be able to parse those things and return them to the caller, for the caller to log it?

Reply:
Got it, they are not very important information, so I think I will remove the print.